components/Picker.jsx

```diff
@@ -0,0 +1,221 @@
+/**
+ * Picker.jsx
+ * Created by Kevin Li 8/16/17
+ */
+
+import React, { createRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+require('../styles/components/_picker.scss');
+
+const fontAwesomeIconId = "usa-dt-picker__button-icon--svg";
+
+const propTypes = {
+    sortFn: PropTypes.func,
+    icon: PropTypes.string,
+    altText: PropTypes.string,
+    iconColor: PropTypes.string,
+    backgroundColor: PropTypes.string,
+    iconSize: PropTypes.string,
+    selectedOption: PropTypes.string,
+    className: PropTypes.string,
+    id: PropTypes.string,
+    options: PropTypes.arrayOf(PropTypes.shape({
+        name: PropTypes.string,
+        onClick: PropTypes.func
+    })),
+    borderType: PropTypes.oneOf(['none', 'bottom', 'full']),
+    dropdownDirection: PropTypes.oneOf(['left', 'right']),
+    borderColor: PropTypes.string,
+    isFixedWidth: PropTypes.bool,
+    children: PropTypes.node
+};
+
+const getBorderType = (borderType, borderColor) => {
+    if (borderType === 'bottom') return { borderBottom: `1px solid ${borderColor}` };
+    if (borderType === 'full') return { border: `1px solid ${borderColor}` };
+    // for custom border styles, pass 'none' and use an id/class to override styles
+    return {};
+};
+
+const pickerRef = createRef();
+const buttonRef = createRef();
+
+const Picker = ({
+    sortFn = () => 'default',
+    selectedOption,
+    options,
+    icon = 'calendar-alt',
+    altText = 'Fiscal Year',
+    iconColor = 'white',
+    iconSize = 'lg',
+    borderType = 'bottom',
+    borderColor = 'white',
+    className = '',
+    id = '',
+    backgroundColor = 'white',
+    isFixedWidth = false,
+    children,
+    dropdownDirection = 'right'
+}) => {
+    const [expanded, setExpanded] = useState(false);
+    const [dimensions, setDimensions] = useState({
+        top: 0,
+        width: 0,
+        left: 0,
+        right: 0
+    });
+
+    const toggleMenu = (e) => {
+        e.preventDefault();
+        setExpanded(!expanded);
+    };
+
+    const defaultSort = (a, b) => {
+        // if no sort fn is provided, sort active element to lowest index
+        if (a === selectedOption) return -1;
+        if (b === selectedOption) return 1;
+        // then, sort alphabetically
+        if (a < b) return -1;
+        if (a > b) return 1;
+        return 0;
+    };
+
+    const handleSort = (a, b) => {
+        if (sortFn(a, b) === 'default') return defaultSort(a, b);
```

Comment:
couldnt we just set the default to `sortFn = () => 'defaultSort(a, b)'` and just return the `sortFn`?

Reply:
What an idea....

Reply:
wait, actually no because the default sort requires `selectedOption` to be inside the scope of the function.

Reply:
wait.... no you're right.